src/main/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/notification/ClaimantResponseAgreedRepaymentRespondentNotificationHandler.java

```diff
@@ -0,0 +1,134 @@
+        return Map.of(
+            callbackKey(ABOUT_TO_SUBMIT),
+            this::notifyRespondent1ForAgreedRepayment
+        );
+    }
+
+    @Override
+    public String camundaActivityId(CallbackParams callbackParams) {
+        return TASK_ID;
+    }
+
+    @Override
+    public List<CaseEvent> handledEvents() {
+        return EVENTS;
+    }
+
+    private CallbackResponse notifyRespondent1ForAgreedRepayment(CallbackParams callbackParams) {
+        CaseData caseData = callbackParams.getCaseData();
+
+        if ((isRespondentSolicitorRegistered(caseData)
+            && caseData.getRespondentSolicitor1EmailAddress() == null)
+            || ((!isRespondentSolicitorRegistered(caseData)
+            && caseData.getRespondent1().getPartyEmail() == null
+            ))) {
+            return AboutToStartOrSubmitCallbackResponse.builder().build();
+        }
+
+        notificationService.sendMail(
+            addEmail(caseData),
+            addTemplate(caseData),
+            addProperties(caseData),
+            String.format(REFERENCE_TEMPLATE, caseData.getLegacyCaseReference())
+        );
+        return AboutToStartOrSubmitCallbackResponse.builder().build();
+    }
+
+    @Override
+    public Map<String, String> addProperties(CaseData caseData) {
+        if (isRespondentSolicitorRegistered(caseData)) {
+            return Map.of(
+                CLAIM_DEFENDANT_LEGAL_ORG_NAME_SPEC, getRespondentLegalOrganizationName(
+                    caseData.getRespondent1OrganisationPolicy().getOrganisation().getOrganisationID()),
+                CLAIM_REFERENCE_NUMBER, caseData.getLegacyCaseReference()
+            );
+        }
+        if (isRespondentNotRepresented(caseData)) {
+            return Map.of(
+                RESPONDENT_NAME, getPartyNameBasedOnType(caseData.getRespondent1()),
+                FRONTEND_URL, pipInPostConfiguration.getCuiFrontEndUrl(),
+                CLAIM_REFERENCE_NUMBER, caseData.getLegacyCaseReference()
+            );
+        }
+        return null;
+    }
+
+    private String addEmail(CaseData caseData) {
+        if (isRespondentSolicitorRegistered(caseData)) {
+            return caseData.getRespondentSolicitor1EmailAddress();
+        }
+        if (isRespondentNotRepresented(caseData)) {
+            return caseData.getRespondent1().getPartyEmail();
+        }
+        return null;
+    }
+
+    private String addTemplate(CaseData caseData) {
+        if (isRespondentSolicitorRegistered(caseData)) {
+            return notificationsProperties.getRespondentSolicitorCcjNotificationTemplate();
+        }
+        if (isRespondentNotRepresented(caseData)) {
+            return notificationsProperties.getRespondentCcjNotificationTemplate();
+        }
+        return null;
+    }
+
+    public String getRespondentLegalOrganizationName(String id) {
+        Optional<Organisation> organisation = organisationService.findOrganisationById(id);
+        String respondentLegalOrganizationName = null;
```

Comment:
You can do this here rather than use a local variable:
`return Optional.ofNullable(organisation).map(Organisation::getName).orElse(null);`

Reply:
Updated with suggestion